galpy/potential_src/potential_c_ext/SpiralArmsPotential.c

```diff
@@ -0,0 +1,763 @@
+#include <math.h>
+#include <galpy_potentials.h>
+
+double gam(double R, double phi, double N, double phi_ref, double r_ref, double tan_alpha);
+
+double dgam_dR(double R, double N, double tan_alpha);
+
+double K(double R, double n, double N, double sin_alpha);
+
+double B(double R, double H, double n, double N, double sin_alpha);
+
+double D(double R, double H, double n, double N, double sin_alpha);
+
+double dK_dR(double R, double n, double N, double sin_alpha);
+
+double dB_dR(double R, double H, double n, double N, double sin_alpha);
+
+double dD_dR(double R, double H, double n, double N, double sin_alpha);
+
+double SpiralArmsPotentialEval(double R, double z, double phi, double t,
+                               struct potentialArg *potentialArgs) {
+    // Get args
+    double *args = potentialArgs->args;
+
+    int nCs = (int) *args++;
+    double amp = *args++;
+    double N = *args++;
+    double sin_alpha = *args++;
+    double tan_alpha = *args++;
+    double r_ref = *args++;
+    double phi_ref = *args++;
+    double Rs = *args++;
+    double H = *args++;
+    double omega = *args++;
+
+    phi = phi - omega * t;
+
+    double g = gam(R, phi, N, phi_ref, r_ref, tan_alpha);
+
+    // Return value of the potential.
+    double sum = 0;
+
+    int n;
+    for (n = 1; n <= nCs; n++) {
+        double Cn = *args++;
```

Comment:
It's best practice to declare variables *outside* of the loop, so move these ``double`` declarations outside of the loop.